querydsl-jpa/src/main/java/com/querydsl/jpa/HQLTemplates.java

```diff
@@ -28,8 +28,9 @@
  * HQLTemplates extends {@link JPQLTemplates} with Hibernate specific extensions
  *
  * @author tiwe
- *
+ * @deprecated Most likely you want to use {@link Hibernate5Templates} instead
```

Comment:
Do we even still support Hibernate 4? I recall an earlier PR already removed compatibility. If so, we might as well replace this implementation.

Reply:
I wasnt sure if we just target 5 or dropped support for 4 alltogether. When replacing the template we will really break hibernate 4 compatibility.

Notice that on Hibernate 5 the right templates for Hibernate 5 are already selected in JpaProvider. So I think it wont harm to keep the old Hibernate 4 templates in there.